### DISCUSSION TITLE
Upgrade boostrap

### DISCUSSION THREAD

**Discussion by enzofrnt** [encode/django-rest-framework#9337] [Potential Issue] [answered]
Hi, I just see that the static CSS is still using Bootstrap 3.4.1 as we can see on this file :
[Bootstrap v3.4.1 (https://getbootstrap.com/)](https://github.com/encode/django-rest-framework/blob/400b4c54419c5c88542ddd0b97219ad4fa8ee29a/rest_framework/static/rest_framework/css/bootstrap-theme.min.css)

Can we expect an upgrade to a newer version, as bootstrap said on her [website](https://getbootstrap.com/docs/3.4/) that this version is in end of life ?
<img width="1170" alt="image" src="https://github.com/encode/django-rest-framework/assets/63660254/ef50dd8e-e2df-4212-99cc-ae71838ec59f">

Maybe an upgrade to V5 ? I don't know.

Thanks. 
And sorry for the bad English, FR…

**Comment by browniebroke** [comment]
Is there anything specifically broken on the current DRF usage of Bootstrap? Unless this is the case, I would lean towards not touching anything... https://github.com/encode/django-rest-framework/discussions/9270

**Reply by enzofrnt** [reply]
> Is there anything specifically broken on the current DRF usage of Bootstrap? Unless this is the case, I would lean towards not touching anything... https://github.com/encode/django-rest-framework/discussions/9270

I don’t know if it’s a problem, but today, when one wants to modify or extend HTML files coming from Django REST Framework, one finds oneself having to use quite old Bootstrap classes, and I don’t understand why we should be limited to that and not update it. Since this version is quite old, the risks of the emergence of vulnerabilities are greater.

**Reply by enzofrnt** [reply]
so ?

**Reply by lovelydinosaur** [reply, marked as answer]
~~It probably is okay for us to accept *some* amount of upgrade changes, we just need a really high quality bar before accepting them.~~ **Edit**: It's unclear to me to what extent we should consider allowing upgrade changes, given that they also introduce friction and risk. If we *are* going to do so, then we need a really high quality bar before accepting them.

...

From https://github.com/encode/django-rest-framework/pull/7411 this looks to be a seriously involved change...

<img width="411" alt="image" src="https://github.com/encode/django-rest-framework/assets/647359/1b273de4-9914-4327-a9bd-2adaebd243d8">

I think that's less risky than some much smaller PRs that effect the Python codebase, since lots of it is just changes in CSS class names, etc. But, still.

Anyways, questions...

* Does the browse-able API look much different after an upgrade, and do we prefer it?
* What implications are there for existing users who have customised some of the UI, and what approach would we suggest if they wanted to retain older behaviours?
* Are we confidently we're able to sensibly review such a large change? (Or are there ways we can introduce it, but with a lower change footprint?)

**Reply by enzofrnt** [reply]
Your concern about the current version we're using is valid; however, it's important to recognize the substantial improvements that have been made from Bootstrap 3 to Bootstrap 5. The latest version offers a more modern approach to design and usability. It introduces new features and optimizations that enhance both the speed and responsiveness of web projects. Additionally, Bootstrap 5 comes with improved accessibility features and drops support for older, less secure browsers, which is crucial for maintaining the security and efficiency of our site. Upgrading to Bootstrap 5 not only future-proofs our work but also significantly improves the user experience with cleaner, more efficient code, and a range of new components and utilities that are not available in Bootstrap 3.

Considering the advantages outlined, it’s clear that they outweigh the potential challenges we might face during the update process. Bootstrap 5’s modern features, enhanced security measures, and improved accessibility support present a compelling case for transition. Importantly, we must also acknowledge that Bootstrap has officially announced the end-of-life status for our current version. This means that continuing with the outdated version not only hinders our access to new functionalities and optimizations but also exposes us to security vulnerabilities due to the lack of future patches or support. Given these factors, transitioning to Bootstrap 5 becomes not just an opportunity for improvement but a necessary step to ensure the longevity and security of our project.

**Reply by enzofrnt** [reply]
So ?
@tomchristie @browniebroke

**Reply by browniebroke** [reply]
I don't see a compelling reason to justify the migration effort/risk. Only "potential" issues. Granted, it's deprecated, but I would still be in the "if ain't broke, don't fix it" camp.

**Reply by enzofrnt** [reply]
So let's use Something deprecated nice ! Modern, secure, alive, updated, nice.
